docs/usage.rst

```diff
@@ -2,4 +2,213 @@
 Usage
 =====
 
-…
+:ref:`Given an HTML form <form>`:
+
+.. _fromstring-example:
+
+>>> from lxml.html import fromstring
+>>> html = b"""<form><input type="hidden" name="foo" value="bar" /></form>"""
+>>> root = fromstring(html, base_url="https://example.com")
+>>> form = root.xpath("//form")[0]
+
+You can use :func:`~form2request.request_from_form` to generate :ref:`form
+submission request data <request>`:
+
+>>> from form2request import request_from_form
+>>> request_from_form(form)
+Request(url='https://example.com?foo=bar', method='GET', headers=[], body=b'')
+
+:func:`~form2request.request_from_form` supports :ref:`user-defined form data
+<data>` and :ref:`choosing a specific form submission button (or none)
+<click>`.
+
+
+.. _form:
+
+Getting a form
+==============
+
+:func:`~form2request.request_from_form` requires an
+:class:`lxml.html.FormElement` object.
```

Comment:
I wonder if it could be convenient if we support parsel.Selector directly in the library, and maybe even SelectorList. 

For example, it may simplify the usage examples in the docs, where form... alternatives can be shown using Selectors first; only then, we may document some of the raw LXML usages.

+1 to drop form... options.